SimplenoteTests/NoticeContorllerTests.swift

```diff
@@ -1,33 +1,142 @@
+        XCTAssertTrue(presentedNoticeView.handler == nil)
+    }
+
+    func testAppendToQueueIfNew() {
+        let noticeA = Notice(message: "Message A", action: nil)
+        let noticeB = Notice(message: "Message B", action: nil)
+
+        controller.present(noticeA)
+        XCTAssertEqual(controller.pendingNotices, 0)
+
+        controller.present(noticeA)
+        XCTAssertEqual(controller.pendingNotices, 0)
 
-    override func setUpWithError() throws {
-        // Put setup code here. This method is called before the invocation of each test method in the class.
+        controller.present(noticeB)
+        XCTAssertEqual(controller.pendingNotices, 1)
+
+        controller.present(noticeB)
+        XCTAssertEqual(controller.pendingNotices, 1)
     }
 
-    override func tearDownWithError() throws {
-        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    func testControllerPresent() {
+        let expectation = XCTestExpectation(description: "Did dismiss notice")
+        let timer = Timer.scheduledTimer(withTimeInterval: TimeInterval(0.5), repeats: false) { (_) in
+            expectation.fulfill()
+        }
+
+        let notice = Notice(message: "Message", action: nil)
+        controller.present(notice, withTimer: timer)
+
+        wait(for: [expectation], timeout: TimeInterval(3))
+
     }
 
-    func testExample() throws {
-        // This is an example of a functional test case.
-        // Use XCTAssert and related functions to verify your tests produce the correct results.
+    func testMakeNoticeView() {
+        let notice = Notice(message: "Message", action: nil)
+
+        controller.present(notice)
+        XCTAssertEqual(notice.message, presenter.noticeView?.message)
+        XCTAssertNil(presenter.noticeView?.handler)
     }
 
-    func testPerformanceExample() throws {
-        // This is an example of a performance test case.
-        self.measure {
-            // Put the code you want to measure the time of here.
+    func testMakeNoticeViewWithAction() {
+        let action = NoticeAction(title: "Title") {
+            print("Action")
         }
+        let notice = Notice(message: "Message ", action: action)
+
+        controller.present(notice)
+        XCTAssertEqual(notice.message, presenter.noticeView?.message)
+        XCTAssertEqual(notice.action?.title, presenter.noticeView?.actionTitle)
+        XCTAssertNotNil(presenter.noticeView?.handler)
+    }
+
+    func testMakeNoticeViewEmptyNotice() {
+        let action = NoticeAction(title: "") {
+            print("")
+        }
+        let notice = Notice(message: "", action: action)
+
+        controller.present(notice)
+        XCTAssertEqual(notice.message, presenter.noticeView?.message)
+        XCTAssertNotEqual(notice.action?.title, presenter.noticeView?.actionTitle)
+        XCTAssertNotNil(presenter.noticeView?.handler)
     }
```

Comment:
`testMakeNoticeViewWithAction` and `testMakeNoticeViewEmptyNotice` are the same except the message. Maybe we can extract a common method?